crypto/evp/e_aes.c

```diff
@@ -3519,6 +3520,13 @@ static int aes_xts_cipher(EVP_CIPHER_CTX *ctx, unsigned char *out,
             || len < AES_BLOCK_SIZE)
         return 0;
 
+    /* NIST's SP 800-38E mandates a 2^20 block limit for XTS mode.
+     * The original IEEE Std 1619-2007 has this as a SHOULD NOT,
+     * whereas the revised IEEE Std 1619-2018 specifies SHALL NOT.
```

Comment:
Such a limit certainly makes sense. Nevertheless, just to be sure, I have to ask: Wouldn't that be a breaking change from 1.1.1 to 3.0, if you enforce this limit for all applications, not only those running with FIPS enabled?

Reply:
Maybe it would even make sense to collect these limits in a common header file?

Reply:
The IEEE suggested the limit originally and made it mandatory in the revised standard.  Agreed, NIST did mandate the limit first and I originally had FIPS_MODE conditionals in place but once I saw the IEEE standard, I removed them.  Yes, it is a breaking change but we do do these when security is impacted as is the case here.


I'll switch the block size check over, yours is a better way.  Thanks for that.

Reply:
Nit, comment format wrong.